### ISSUE TITLE
Gradle fails on openjdk-13

### ISSUE THREAD

**Issue by rickardp** [gradle/gradle#8681]
### Expected Behavior
Gradle fails to start on openjdk-13.

### Current Behavior
When starting Gradle, the following error occurs
```
FAILURE: Build failed with an exception.

* Where:
Settings file '/src/settings.gradle'

* What went wrong:
Could not compile settings file '/src/settings.gradle'.
> startup failed:
  General error during semantic analysis: Unsupported class file major version 57

  java.lang.IllegalArgumentException: Unsupported class file major version 57
        at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:184)
        at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
        at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:152)
        at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:273)
        at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:81)
        at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:254)
        at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:192)
        at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:172)
        at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:128)
        at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveClassNullable(AsmReferenceResolver.java:59)
        at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveClass(AsmReferenceResolver.java:46)
        at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveNonArrayType(AsmReferenceResolver.java:81)
        at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveType(AsmReferenceResolver.java:72)
        at org.codehaus.groovy.ast.decompiled.MemberSignatureParser.createMethodNode(MemberSignatureParser.java:55)
        at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.lazyInitMembers(DecompiledClassNode.java:195)
```

### Context
I had to use openjdk-13, since version 11 does not support Alpine. I am trying to get Gradle running in an Alpine-based docker image.

### Steps to Reproduce (for bugs)
* Use official openjdk:13-alpine image
* Install Gradle from tarball
* Run Gradle on minimal project

### Your Environment
Build scan fails with the same exception as above.

**Comment by rickardp** [comment]
Workaround is to keep a separate JDK 8 installation to run Gradle on (this is the only version except for 13 to run on Alpine, because this is what Oracle provided)

**Comment by sormuras** [comment]
Same same for JUnit's build on [OpenJDK-13-ea](https://dev.azure.com/junit-team/junit5/_build/results?buildId=45&view=logs&jobId=a2af646c-3471-5c61-2ba1-e88936f3c0bd&taskId=0702fac9-1bea-55ad-c910-65c92b03e8d6&lineStart=102&lineEnd=103&colStart=1&colEnd=1).

**Comment by marcphilipp** [comment]
Looks like that needs Groovy 2.5.7 which isn't released, yet:
https://issues.apache.org/jira/browse/GROOVY-9020

**Comment by aguibert** [comment]
@marcphilipp FYI looks like Groovy 2.5.7 was just released a few days ago: https://github.com/apache/groovy/releases/tag/GROOVY_2_5_7

**Comment by szpak** [comment]
@aguibert Nope, it just a release tag. It's still [not available](https://repo1.maven.org/maven2/org/codehaus/groovy/groovy/) in Maven Centra.

**Comment by aguibert** [comment]
@szpak I see it here: https://repo1.maven.org/maven2/org/codehaus/groovy/groovy/2.5.7/
It very may well have been added within the last 2 hours though

**Comment by szpak** [comment]
@marcphilipp Has the upgrade to Groovy 2.5.7 unblocked this issue or there is something more to be done on the Groovy side to be able to use Gradle with Java 13?

**Comment by aguibert** [comment]
hi @marcphilipp, any outlook on when Java 13 support will be added? Java 13 entered the rampdown phase recently so it should be stable enough at this point that it's worth kicking the tires.

**Comment by marcphilipp** [comment]
@paplorinc @big-guy Can you provide a timeline for that?

**Comment by paplorinc** [comment]
The Groovy blocker was solved in their latest snapshot, maybe we could try it out.
@big-guy, will you take over?

**Comment by snicoll** [comment]
Can anyone provide a timeline for a fix? We'd like to validate that the upcoming Spring Framework 5.2 release builds fine with Java 13.

**Comment by paplorinc** [comment]
Current plans are to target Gradle 6.0 with this, since in Groovy 2.5.4-2.5.7 a few regressions were introduced, i.e.:
 * https://issues.apache.org/jira/browse/GROOVY-9167
 * https://issues.apache.org/jira/browse/GROOVY-9166
 * https://issues.apache.org/jira/browse/GROOVY-9171
+ some performance regressions that our benchmarks have caught.

We're waiting for Groovy 2.5.8 to be released, and see if they were addressed.

cc: @paulk-asert

**Comment by AlanBateman** [comment]
Same thing happens with the recently released Groovy 2.5.8:

```
$ groovy-2.5.8/bin/groovy hello.groovy 
java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7
	at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:43)
	at org.codehaus.groovy.reflection.GroovyClassValueFactory.<clinit>(GroovyClassValueFactory.java:35)
:
```

With JDK 13, 12, 11, ... then it seems to load org.codehaus.groovy.vmplugin.v9.Java9 but then switch to org.codehaus.groovy.vmplugin.v7.Java7 where it tries to hack into one of Lookup's private constructors. That causes a warning at run-time. The constructor doesn't exist in JDK 14 so the hack fails.

**Comment by paplorinc** [comment]
@big-guy, now that the latest Gradle nightly includes the Groovy upgrade, what else do we need for JDK 13 compatibility?

**Comment by jabrena** [comment]
Any progress with this issue?

I am using latest `Intellij` with latest `Gradle` wrapper (5.6) and I receive the following error when I compile a Java class using a feature included in Java 13:

```
> startup failed:
  General error during semantic analysis: Unsupported class file major version 57
```

**Comment by paplorinc** [comment]
@jabrena, the JDK 13 copatibility changes are merged in the latest 6.0 snapshot, could you try it out?
cc: @big-guy

**Comment by yschimke** [comment]
Working well for me

https://github.com/square/okhttp/issues/5213
https://circleci.com/gh/square/okhttp/4949?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Two issues
1) related to JDK 13, out of scope
2) A spotless bug I've reported as a courtesy https://github.com/diffplug/spotless/issues/440

**Comment by jabrena** [comment]
Hi @paplorinc, today, I will test it.

Many thanks :)

Cheers

Juan Antonio

**Comment by jabrena** [comment]
Hi @paplorinc, where is located the 6.0 snapshot?
Here, I don´t see that version: https://services.gradle.org/distributions/

**Comment by akarnokd** [comment]
`distributions-snapshots`, https://services.gradle.org/distributions-snapshots/gradle-6.0-20190821220044+0000-all.zip

[Works](https://travis-ci.org/akarnokd/RxJava3_BuildMatrix/builds/575458795) for me with JDK 13 but not with JDK 14 sadly.

**Comment by jabrena** [comment]
Many thanks @akarnokd, I go now :)

**Comment by cocorossello** [comment]
Works fine for us with jdk-13 b33. Thanks

**Comment by big-guy** [comment]
This will be in 6.0

**Comment by ldco2016** [comment]
Wow this is super confusing, what is the verdict here? How do I solve `General error during semantic analysis: Unsupported class file major version 57`?

**Comment by aguibert** [comment]
@ldco2016 you need to upgrade to Gradle 6.0 or newer, which will support JDK 13

**Comment by ldco2016** [comment]
@aguibert , I know to go to `gradle-wrapper.properties` and change `distributionUrl` to this: `gradle-6.0-all.zip` but what do I change the `classpath()` to in `build.gradle`? It is currently, `com.android.tools.build:gradle:3.5.2`

**Comment by rychter** [comment]
As it was mentioned, changing distributionUrl to: gradle-6.0-all.zip in gradle-wrapper.properties fixed the issue.

**Comment by Jaswanth75** [comment]
@aguibert  I'm recently trying to migrate from grails 2.x to grails 4.0.4 and using openjdk 13. I'm facing few issues with gradle build. I'm using gradle 6.6.1.
Issue: `General error during semantic analysis: Unsupported class file major version 57`. Please help me fix this
